src/js/utils/template-processor.js

```diff
@@ -23,33 +23,47 @@ const operators = {
   '??': (value, defaults) => value ?? defaults,
 };
 
-class PartialDirective {
+class PartialTemplate {
   constructor(template) {
     this.template = template;
   }
 }
 
+const Directives = {
+  partial: (expression, part, state) => {
+    state[expression] = new PartialTemplate(part.template);
+  },
+  if: (expression, part, state) => {
+    if (evaluateCondition(part.expression, state)) {
+      part.replace(new TemplateInstance(part.template, state, processor));
+    } else {
+      part.replace('');
+    }
+  },
+};
+
+const DirectiveNames = Object.keys(Directives);
+
+export function transformDirectiveAliases(template) {
+  // Transform short-hand if/partial templates to directive & expression.
+  const query = DirectiveNames.map((name) => `template[${name}]`).join(',');
+  template.content.querySelectorAll(query).forEach((templateEl) => {
+    const directive = DirectiveNames.find((n) => templateEl.hasAttribute(n));
+    if (directive) {
+      templateEl.setAttribute('directive', directive);
+      templateEl.setAttribute('expression', templateEl.getAttribute(directive));
+    }
+  });
+  return template;
+}
+
 export const processor = {
   processCallback(instance, parts, state) {
     if (!state) return;
 
     for (const [expression, part] of parts) {
-      if (part instanceof InnerTemplatePart) {
-        switch (part.directive) {
-          case 'partial': {
-            state[expression] = new PartialDirective(part.template);
-            break;
-          }
-          case 'if':
-            if (evaluateCondition(part.expression, state)) {
-              part.replace(
-                new TemplateInstance(part.template, state, processor)
-              );
-            } else {
-              part.replace('');
-            }
-            break;
-        }
+      if (part.directive && part instanceof InnerTemplatePart) {
```

Comment:
😎 I like this.

Reply:
question(non-blocking): Should this conditional order be switched?

Reply:
I imagined a property lookup is faster than an instanceof check but not sure.
yes, I can see why it makes a little more sense in an ordering sense to first check instanceof but perf :)